### PR TITLE
fix TimeToClose shows zero, resolve #578

### LINKED ISSUE #578: TimeToClose is calculated incorrectly if at least 3 runners had not finished
`Time to close` is calculated using the time of the 3rd runner. If there are not at least 3 runners in the finish, `time to close` can be incorrectly calculated and shown as 0.

https://github.com/Quick-Event/quickbox/blob/515f64d87bc02a2ded6e99e1e2cd5cb463078308/quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp#L169-L185

### PR DESCRIPTION
**Pull request by otahirs**
right now solves two issues.
1) 3rd best time was incorrectly calculated even if finished runners count was less than 3 as `MAX(TOP 3)`
2) TimeToClose should now display 0 if all runners finished (even if there are less than 3), display nothing if there is no enough information to do the calculation or display calculated timeToClose down to 0.

### CODE REVIEW

File: quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp [hunk 1]
```diff
@@ -171,30 +171,31 @@ QVariant EventStatisticsModel::value(int row_ix, int column_ix) const
...
+			return QVariant();
 		}
+		int start_last_msec = value(row_ix, col_startLastMs).toInt();
+		int time3_msec = value(row_ix, col_time3Ms).toInt();
+		int time_to_close_msec = stage_start_msec + start_last_msec + time3_msec - curr_time_msec;
+		if(time_to_close_msec < 0)
+			time_to_close_msec = 0;
+		return time_to_close_msec;
 	}
-	else if(column_ix == col_runnersNotFinished) {
```
Comment (otahirs) [inline quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp:188]: additionally `else` keywords removed to make the function consistent, they are not needed anyway

File: quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp [hunk 2]
```diff
@@ -171,30 +171,31 @@ QVariant EventStatisticsModel::value(int row_ix, int column_ix) const
 	if(column_ix == col_timeToCloseMs) {
 		int stage_start_msec = eventPlugin()->stageStartMsec(eventPlugin()->currentStageId());
 		int curr_time_msec = QTime::currentTime().msecsSinceStartOfDay();
-		//int runners_count = value(row_ix, col_runnersCount).toInt();
+		int runners_count = value(row_ix, col_runnersCount).toInt();
 		int runners_finished = value(row_ix, col_runnersFinished).toInt();
-		if(runners_finished == 0) {
-			return QVariant();
+		if(runners_count == runners_finished) {
```
Comment (otahirs) [inline quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp:176]: if there are less then 3 runners but all finished, timeToClose will be shown corrently as 0

File: quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp [hunk 3]
```diff
@@ -171,30 +171,31 @@ QVariant EventStatisticsModel::value(int row_ix, int column_ix) const
...
+			return 0;
 		}
-		else {
-			int start_last_msec = value(row_ix, col_startLastMs).toInt();
-			int time3_msec = value(row_ix, col_time3Ms).toInt();
-			int time_to_close_msec = stage_start_msec + start_last_msec + time3_msec - curr_time_msec;
-			if(time_to_close_msec < 0)
-				time_to_close_msec = 0;
-			return time_to_close_msec;
+		if(runners_finished < 3) {
```
Comment (otahirs) [inline quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp:179]: if there are not at least 3 finished runners, time to close will stay blank

File: quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp [hunk 4]
```diff
@@ -171,30 +171,31 @@ QVariant EventStatisticsModel::value(int row_ix, int column_ix) const
...
+			return QVariant();
 		}
+		int start_last_msec = value(row_ix, col_startLastMs).toInt();
```
Comment (otahirs) [inline quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp:182]: unchanged

File: quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp [hunk 5]
```diff
@@ -171,30 +171,31 @@ QVariant EventStatisticsModel::value(int row_ix, int column_ix) const
...
 		}
-		else {
-			int start_last_msec = value(row_ix, col_startLastMs).toInt();
-			int time3_msec = value(row_ix, col_time3Ms).toInt();
-			int time_to_close_msec = stage_start_msec + start_last_msec + time3_msec - curr_time_msec;
-			if(time_to_close_msec < 0)
-				time_to_close_msec = 0;
-			return time_to_close_msec;
+		int time3_msec = value(row_ix, col_time3Ms).toInt();
+		if(time3_msec == 0) {
```
Comment (otahirs) [inline quickevent/app/plugins/Runs/src/eventstatisticswidget.cpp:180]: if there are not at least 3 finished runners, time to close will stay blank

**Comment by fvacek** [comment]
I've merged pull requests in wrong order, please can you resolve the conflict.

**Pull request merged**